core/capabilities/ccip/ccipaptos/extradatadecoder.go

```diff
@@ -0,0 +1,201 @@
+type ExtraDataDecoder struct{}
+
+var _ ccipcommon.ExtraDataDecoder = (*ExtraDataDecoder)(nil)
+
+const (
+	aptosDestExecDataKey = "destGasAmount"
+)
+
+var (
+	// bytes4 public constant EVM_EXTRA_ARGS_V1_TAG = 0x97a657c9;
+	evmExtraArgsV1Tag = hexutil.MustDecode("0x97a657c9")
+
+	// bytes4 public constant GENERIC_EXTRA_ARGS_V2_TAG = 0x181dcf10;
+	genericExtraArgsV2Tag = hexutil.MustDecode("0x181dcf10")
+
+	// bytes4 public constant SVM_EXTRA_EXTRA_ARGS_V1_TAG = 0x1f3b3aba
+	svmExtraArgsV1Tag = hexutil.MustDecode("0x1f3b3aba")
+
+	uint32Type     = mustNewType("uint32")
+	uint64Type     = mustNewType("uint64")
+	uint256Type    = mustNewType("uint256")
+	boolType       = mustNewType("bool")
+	bytes32Type    = mustNewType("bytes32")
+	bytes32ArrType = mustNewType("bytes32[]")
+
+	// Arguments for decoding destGasAmount
+	destGasAmountArguments = abi.Arguments{
+		{Name: aptosDestExecDataKey, Type: uint32Type},
+	}
+
+	// Arguments matching the fields of EVMExtraArgsV1 struct
+	evmExtraArgsV1Fields = abi.Arguments{
+		{Name: "gasLimit", Type: uint256Type},
+	}
+
+	// Arguments matching the fields of GenericExtraArgsV2 struct
+	genericExtraArgsV2Fields = abi.Arguments{
+		{Name: "gasLimit", Type: uint256Type},
+		{Name: "allowOutOfOrderExecution", Type: boolType},
+	}
+
+	// Arguments matching the fields of SVMExtraArgsV1 struct
+	svmExtraArgsV1Fields = abi.Arguments{
+		{Name: "computeUnits", Type: uint32Type},
+		{Name: "accountIsWritableBitmap", Type: uint64Type},
+		{Name: "allowOutOfOrderExecution", Type: boolType},
+		{Name: "tokenReceiver", Type: bytes32Type},
+		{Name: "accounts", Type: bytes32ArrType},
+	}
+)
+
+func mustNewType(typeStr string) abi.Type {
+	t, err := abi.NewType(typeStr, "", nil)
+	if err != nil {
+		panic(fmt.Sprintf("failed to create ABI type %s: %v", typeStr, err))
+	}
+	return t
+}
+
+// DecodeDestExecDataToMap reformats bytes into a chain agnostic map[string]interface{} representation for dest exec data
+func (d ExtraDataDecoder) DecodeDestExecDataToMap(destExecData cciptypes.Bytes) (map[string]any, error) {
+	args := make(map[string]any)
+	err := destGasAmountArguments.UnpackIntoMap(args, destExecData)
+	if err != nil {
+		if len(destExecData) != 32 {
+			return nil, fmt.Errorf("decode dest gas amount: expected 32 bytes for uint32, got %d: %w", len(destExecData), err)
+		}
+
+		var val big.Int
+		val.SetBytes(destExecData)
+		if val.Cmp(big.NewInt(0xFFFFFFFF)) > 0 {
+			return nil, fmt.Errorf("decode dest gas amount: value %s exceeds uint32 max: %w", val.String(), err)
+		}
+
+		return nil, fmt.Errorf("decode dest gas amount: %w", err)
+	}
+
+	if _, ok := args[aptosDestExecDataKey]; !ok {
+		return nil, fmt.Errorf("failed to unpack key '%s' for dest gas amount", aptosDestExecDataKey)
+	}
+
+	return args, nil
+}
+
+// DecodeExtraArgsToMap reformats bytes into a chain agnostic map[string]any representation for extra args
+func (d ExtraDataDecoder) DecodeExtraArgsToMap(extraArgs cciptypes.Bytes) (map[string]any, error) {
```

Comment:
This code is functionally identical to the ExtraDataDecoder for EVM, but I'm confused as to why, since this would be used to decode extraArgs provided on Aptos as a source. Do we use EVM-style ABI-encoding for Aptos?